### PR TITLE
fix(cli): do not throw if the source could not be resolved

### PR DESCRIPTION
**Pull request by armandobelardo**

### CODE REVIEW

File: packages/cli/generation/ir-generator/src/resolvers/SourceResolver.ts [hunk 1]
```diff
@@ -10,14 +11,19 @@ export interface SourceResolver {
...
     }) => Promise<ResolvedSource | undefined>;
-    resolveSourceOrThrow: (args: { source: RawSchemas.SourceSchema; file: FernFileContext }) => Promise<ResolvedSource>;
+    resolveSourceOrThrow: (args: {
+        source: RawSchemas.SourceSchema;
+        file: FernFileContext;
+    }) => Promise<ResolvedSource | undefined>;
 }
 
 export class SourceResolverImpl implements SourceResolver {
+    private readonly taskContext: TaskContext;
```
Comment (amckinney) [inline packages/cli/generation/ir-generator/src/resolvers/SourceResolver.ts:21]: nit: Can we name this property `context` for consistency with elsewhere? That way we just say:

```typescript
this.context.logger.warn(`Cannot resolve source ${source.openapi} from file ${file.relativeFilepath}`);
```

**Review by amckinney — APPROVED** [review]

File: packages/cli/generation/remote-generation/remote-workspace-runner/src/createAndStartJob.ts [hunk 2]
```diff
@@ -148,6 +149,21 @@ async function createJob({
                 RelativeFilePath.of(PROJECT_CONFIG_FILENAME)
             );
             await writeFile(absolutePathToFernConfigJson, JSON.stringify(projectConfig.rawConfig, undefined, 2));
+            // write sources
+            const sources = workspace.getSources();
+            for (const source of sources) {
+                context.logger.debug(`[TEST] Writing source ${source.absoluteFilePath}`);
```
Comment (dsinghvi) [inline packages/cli/generation/remote-generation/remote-workspace-runner/src/createAndStartJob.ts:155]: nit: lets delete this

File: packages/cli/generation/remote-generation/remote-workspace-runner/src/createAndStartJob.ts [hunk 3]
```diff
@@ -148,6 +149,21 @@ async function createJob({
...
+                const sourceContents = await readFile(source.absoluteFilePath);
+                const relativeLocation = relative(workspace.absoluteFilepath, source.absoluteFilePath);
```
Comment (dsinghvi) [inline packages/cli/generation/remote-generation/remote-workspace-runner/src/createAndStartJob.ts:157]: What does `relative` return if the source absolute filepath is outside of the fern folder?

Comment (dsinghvi) [reply packages/cli/generation/remote-generation/remote-workspace-runner/src/createAndStartJob.ts:157]: I wonder if we need to check the returned result to make sure lines 158-162 dont throw

Comment (armandobelardo) [reply packages/cli/generation/remote-generation/remote-workspace-runner/src/createAndStartJob.ts:157]: I've wrapped it all in a try catch

**Pull request merged**